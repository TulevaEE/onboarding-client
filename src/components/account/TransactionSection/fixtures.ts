import { Transaction } from '../../common/apiModels';

export const contribution: Transaction = {
  amount: 313.57,
  currency: 'EUR',
  time: '2023-01-23T15:12:58Z',
  isin: 'EE3600109435',
};

export const subtraction: Transaction = {
  amount: -100.0,
  currency: 'EUR',
  time: '2023-01-24T16:20:55Z',
  isin: 'EE3600001707',
};
