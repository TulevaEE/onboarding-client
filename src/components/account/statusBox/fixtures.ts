import { SourceFund, UserConversion } from '../../common/apiModels';

export const incompleteConversion: UserConversion = {
  secondPillar: {
    selectionComplete: false,
    selectionPartial: false,
    transfersComplete: false,
    transfersPartial: false,
    paymentComplete: false,
    pendingWithdrawal: false,
    contribution: { yearToDate: 200, lastYear: 40, total: 250 },
    subtraction: { yearToDate: 0, lastYear: 0, total: 0 },
  },
  thirdPillar: {
    selectionComplete: false,
    selectionPartial: false,
    transfersComplete: false,
    transfersPartial: false,
    paymentComplete: false,
    pendingWithdrawal: false,
    contribution: { yearToDate: 300, lastYear: 50, total: 450 },
    subtraction: { yearToDate: 0, lastYear: 0, total: 0 },
  },
};

export const completeConversion: UserConversion = {
  secondPillar: {
    selectionComplete: true,
    selectionPartial: true,
    transfersComplete: true,
    transfersPartial: true,
    paymentComplete: true,
    pendingWithdrawal: false,
    contribution: { yearToDate: 560, lastYear: 100, total: 20600 },
    subtraction: { yearToDate: 0, lastYear: 0, total: 0 },
  },
  thirdPillar: {
    selectionComplete: true,
    selectionPartial: true,
    transfersComplete: true,
    transfersPartial: true,
    paymentComplete: true,
    pendingWithdrawal: false,
    contribution: { yearToDate: 2500, lastYear: 1000, total: 37000 },
    subtraction: { yearToDate: 0, lastYear: 0, total: 0 },
  },
};

export const completeThirdPillarConversion: UserConversion = {
  secondPillar: {
    selectionComplete: false,
    selectionPartial: false,
    transfersComplete: false,
    transfersPartial: false,
    paymentComplete: false,
    pendingWithdrawal: false,
    contribution: { yearToDate: 560, lastYear: 100, total: 20600 },
    subtraction: { yearToDate: 0, lastYear: 0, total: 0 },
  },
  thirdPillar: {
    selectionComplete: true,
    selectionPartial: true,
    transfersComplete: true,
    transfersPartial: true,
    paymentComplete: true,
    pendingWithdrawal: false,
    contribution: { yearToDate: 2500, lastYear: 1000, total: 37000 },
    subtraction: { yearToDate: 0, lastYear: 0, total: 0 },
  },
};

export const completeSecondPillarConversion: UserConversion = {
  secondPillar: {
    selectionComplete: true,
    selectionPartial: true,
    transfersComplete: true,
    transfersPartial: true,
    paymentComplete: true,
    pendingWithdrawal: false,
    contribution: { yearToDate: 2500, lastYear: 100, total: 37000 },
    subtraction: { yearToDate: 0, lastYear: 0, total: 0 },
  },
  thirdPillar: {
    selectionComplete: false,
    selectionPartial: false,
    transfersComplete: false,
    transfersPartial: false,
    paymentComplete: false,
    pendingWithdrawal: false,
    contribution: { yearToDate: 560, lastYear: 100, total: 20600 },
    subtraction: { yearToDate: 0, lastYear: 0, total: 0 },
  },
};

export const activeSecondPillar: SourceFund = {
  activeFund: true,
  name: 'Tuleva II Samba Pensionifond',
  fundManager: { name: 'Tuleva' },
  pillar: 2,
  managementFeePercent: 0.003,
  isin: 'EE000123',
  price: 23,
  unavailablePrice: 0,
  currency: 'EUR',
  ongoingChargesFigure: 0.0049,
  contributions: 500,
  subtractions: 0,
  profit: 500,
};

export const activeThirdPillar: SourceFund = {
  activeFund: true,
  name: 'Tuleva III Samba Pensionifond',
  fundManager: { name: 'Tuleva' },
  pillar: 3,
  managementFeePercent: 0.003,
  isin: 'EE3600001707',
  price: 23,
  unavailablePrice: 0,
  currency: 'EUR',
  ongoingChargesFigure: 0.0049,
  contributions: 500,
  subtractions: 0,
  profit: 500,
};

export const highFeeThirdPillar: SourceFund = {
  activeFund: true,
  name: 'LHV High Cost Fund III',
  fundManager: { name: 'LHV' },
  pillar: 3,
  managementFeePercent: 0.015,
  isin: 'EE1234567',
  price: 23,
  unavailablePrice: 0,
  currency: 'EUR',
  ongoingChargesFigure: 0.0156,
  contributions: 500,
  subtractions: 0,
  profit: 500,
};

export const highFeeSecondPillar: SourceFund = {
  activeFund: true,
  name: 'LHV High Cost Fund II',
  fundManager: { name: 'LHV' },
  pillar: 2,
  managementFeePercent: 0.015,
  isin: 'EE432432',
  price: 23,
  unavailablePrice: 0,
  currency: 'EUR',
  ongoingChargesFigure: 0.0156,
  contributions: 500,
  subtractions: 0,
  profit: 500,
};
