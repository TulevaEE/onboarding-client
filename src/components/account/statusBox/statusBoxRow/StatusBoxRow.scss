@import '../../../variables';

.status-box-row li {
  list-style: none;
}

.status-box-row ul {
  padding: 0;
  margin-bottom: 0;
}

.status-box-row .btn {
  min-width: 161px;
}

.fa.fa-check {
  color: $green;
}

.fa.fa-times {
  color: $red;
}

.fa.fa-exclamation-triangle {
  color: $yellow-dark;
}
