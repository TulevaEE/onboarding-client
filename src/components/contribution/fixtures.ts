import { ThirdPillarContribution } from '../common/apiModels';

export const contribution: ThirdPillarContribution = {
  time: '2023-01-23T15:12:58Z',
  sender: 'Tuleva Fondid AS',
  amount: 313.57,
  currency: 'EUR',
  pillar: 3,
};
