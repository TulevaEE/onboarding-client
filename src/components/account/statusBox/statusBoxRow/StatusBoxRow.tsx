import React from 'react';
import './StatusBoxRow.scss';

const CheckMark: React.FunctionComponent<{
  checked?: boolean;
  warning?: boolean;
  error?: boolean;
}> = ({ checked = false, warning = false, error = false }) => (
  <div
    className={`ml-3 mr-2 fa${checked ? ' fa-check' : ''}${error ? ' fa-times' : ''}${
      warning ? ' fa-exclamation-triangle' : ''
    }`}
  />
);

export const StatusBoxRow: React.FunctionComponent<{
  name?: React.ReactNode;
  lines?: React.ReactNode[];
  showAction?: boolean;
  ok?: boolean;
  warning?: boolean;
  error?: boolean;
  last?: boolean;
  children?: React.ReactNode;
}> = ({
  name = '',
  lines = [],
  showAction = false,
  ok = false,
  warning = false,
  error = false,
  children = '',
  last = false,
}) => {
  const formattedLines = (
    <ul>
      {lines &&
        lines.length > 0 &&
        lines.map((line, i) => {
          return (
            <li className="pl-2" key={i}>
              {line}
            </li>
          );
        })}
    </ul>
  );
  return (
    <div
      className={`d-flex flex-sm-row flex-column justify-content-between py-2 status-box-row ${
        !last ? 'tv-table__row' : ''
      }`}
    >
      <div className="d-flex">
        <div className="d-flex flex-column justify-content-center">
          <CheckMark checked={ok} warning={warning} error={error} />
        </div>
        <div className="d-flex flex-column justify-content-center">
          <div className="mt-0 pt-1 pl-2">
            <b>{name}</b>
          </div>
          {formattedLines}
        </div>
      </div>
      <div className="d-flex flex-column justify-content-center my-2 mx-3 text-nowrap">
        {showAction && children ? children : ''}
      </div>
    </div>
  );
};

export default StatusBoxRow;
