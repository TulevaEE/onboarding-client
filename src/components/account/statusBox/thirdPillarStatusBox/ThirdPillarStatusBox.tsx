import React from 'react';
import { FormattedMessage } from 'react-intl';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import StatusBoxRow from '../statusBoxRow';
import { Conversion, SourceFund } from '../../../common/apiModels';
import { InfoTooltip } from '../../../common';
import { State } from '../../../../types';
import ThirdPillarPaymentsAmount from './ThirdPillarContributionAmount';

interface Props {
  conversion: Conversion;
  loading: boolean;
  thirdPillarFunds: SourceFund[];
  thirdPillarActive: boolean;
}

export const ThirdPillarStatusBox: React.FunctionComponent<Props> = ({
  conversion,
  loading = false,
  thirdPillarFunds = [],
  thirdPillarActive,
}) => {
  const activeFund = thirdPillarFunds.find((fund) => fund.activeFund);

  if (!thirdPillarActive || !activeFund) {
    return (
      <StatusBoxRow
        error
        showAction={!loading}
        name={<FormattedMessage id="account.status.choice.pillar.third" />}
        lines={[<FormattedMessage id="account.status.choice.pillar.third.missing.label" />]}
      >
        <Link to="/3rd-pillar-flow" className="btn btn-primary">
          <FormattedMessage id="account.status.choice.pillar.third.missing.action" />
        </Link>
      </StatusBoxRow>
    );
  }

  const month = new Date().getMonth();
  const isFebruaryToNovember = month > 0 && month < 11;
  const isDecember = month === 11;

  const isPartiallyConverted = conversion.selectionPartial || conversion.transfersPartial;
  if (!isPartiallyConverted) {
    if (conversion.weightedAverageFee > 0.005) {
      return (
        <StatusBoxRow
          error
          showAction={!loading}
          name={<FormattedMessage id="account.status.choice.pillar.third" />}
          lines={[
            <>
              <FormattedMessage id="account.status.choice.highFee.label" />
              <InfoTooltip name="third-pillar-tooltip">
                <FormattedMessage id="account.status.choice.highFee.description" />
              </InfoTooltip>
            </>,
            <ThirdPillarPaymentsAmount />,
          ]}
        >
          <Link to="/3rd-pillar-flow" className="btn btn-primary">
            <FormattedMessage id="account.status.choice.choose.low.fees" />
          </Link>
        </StatusBoxRow>
      );
    }
    return (
      <StatusBoxRow
        ok
        showAction={!loading}
        name={<FormattedMessage id="account.status.choice.pillar.third" />}
        lines={[
          <>
            <FormattedMessage id="account.status.choice.lowFee.label" />
            <InfoTooltip name="third-pillar-tooltip">
              <FormattedMessage id="account.status.choice.lowFee.description" />
            </InfoTooltip>
          </>,
          <ThirdPillarPaymentsAmount />,
        ]}
      >
        <Link to="/3rd-pillar-flow" className="btn btn-light">
          <FormattedMessage id="account.status.choice.pillar.third.inactive.action" />
        </Link>
      </StatusBoxRow>
    );
  }

  const isFullyConverted = conversion.selectionComplete && conversion.transfersComplete;
  if (!isFullyConverted) {
    return (
      <StatusBoxRow
        warning
        showAction={!loading}
        name={<FormattedMessage id="account.status.choice.pillar.third" />}
        lines={[
          <FormattedMessage id="account.status.choice.pillar.third.transferIncomplete.label" />,
          <ThirdPillarPaymentsAmount />,
        ]}
      >
        <Link to="/3rd-pillar-payment" className="btn btn-primary">
          <FormattedMessage id="account.status.choice.pillar.third.success.action" />
        </Link>
      </StatusBoxRow>
    );
  }

  if (conversion.contribution.total === 0) {
    return (
      <StatusBoxRow
        error
        showAction={!loading}
        name={<FormattedMessage id="account.status.choice.pillar.third" />}
        lines={[
          <>
            <FormattedMessage id="account.status.choice.pillar.third.paymentIncomplete.label" />
            <InfoTooltip name="third-pillar-tooltip">
              <FormattedMessage id="account.status.choice.pillar.third.paymentInfo" />
            </InfoTooltip>
          </>,
          <ThirdPillarPaymentsAmount />,
        ]}
      >
        <Link to="/3rd-pillar-payment" className="btn btn-primary">
          <FormattedMessage id="account.status.choice.pillar.third.success.action" />
        </Link>
      </StatusBoxRow>
    );
  }

  if (conversion.contribution.yearToDate === 0 && isFebruaryToNovember) {
    return (
      <StatusBoxRow
        warning
        showAction={!loading}
        name={<FormattedMessage id="account.status.choice.pillar.third" />}
        lines={[
          <>
            <FormattedMessage id="account.status.choice.pillar.third.thisYearPaymentIncomplete.label" />
            <InfoTooltip name="third-pillar-tooltip">
              <FormattedMessage id="account.status.choice.pillar.third.paymentInfo" />
            </InfoTooltip>
          </>,
          <ThirdPillarPaymentsAmount />,
        ]}
      >
        <Link to="/3rd-pillar-payment" className="btn btn-primary">
          <FormattedMessage id="account.status.choice.pillar.third.success.action" />
        </Link>
      </StatusBoxRow>
    );
  }

  if (conversion.contribution.yearToDate === 0 && isDecember) {
    return (
      <StatusBoxRow
        error
        showAction={!loading}
        name={<FormattedMessage id="account.status.choice.pillar.third" />}
        lines={[
          <FormattedMessage
            id="account.status.choice.pillar.third.paymentDeadline"
            values={{
              b: (chunks: string) => <b className="highlight text-nowrap">{chunks}</b>,
            }}
          />,
          <ThirdPillarPaymentsAmount />,
        ]}
      >
        <Link to="/3rd-pillar-payment" className="btn btn-primary">
          <FormattedMessage id="account.status.choice.pillar.third.success.action" />
        </Link>
      </StatusBoxRow>
    );
  }

  return (
    <StatusBoxRow
      ok
      showAction={!loading}
      name={<FormattedMessage id="account.status.choice.pillar.third" />}
      lines={[
        <>
          <FormattedMessage id="account.status.choice.lowFee.index.label" />
          <InfoTooltip name="third-pillar-tooltip">
            <FormattedMessage id="account.status.choice.lowFee.description" />
          </InfoTooltip>
        </>,
        <ThirdPillarPaymentsAmount />,
      ]}
    >
      <Link to="/3rd-pillar-payment" className="btn btn-outline-primary">
        <FormattedMessage id="account.status.choice.pillar.third.success.action" />
      </Link>
    </StatusBoxRow>
  );
};

const mapStateToProps = (state: State) => ({
  conversion: state.login.userConversion.thirdPillar,
  loading: state.login.loadingUserConversion,
  thirdPillarFunds: state.thirdPillar.sourceFunds,
  thirdPillarActive: state.login.user.thirdPillarActive,
});

export default connect(mapStateToProps)(ThirdPillarStatusBox);
