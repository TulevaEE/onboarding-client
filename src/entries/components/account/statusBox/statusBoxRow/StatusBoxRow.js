import React from 'react';
import { PropTypes as Types } from 'prop-types';
import { Message } from 'retranslate';
import './StatusBoxRow.scss';

const CheckMark = ({ checked }) => (
  <div className={`ml-3 mr-2 fa${checked ? ' fa-check' : ' fa-times'}`} />
);
CheckMark.defaultProps = {
  checked: false,
};

CheckMark.propTypes = {
  checked: Types.bool,
};

const StatusBoxRow = ({ name, lines, showAction, ok, children, last }) => {
  const displayName = <Message>{name}</Message>;
  const formattedLines = (
    <ul className="status-box-row">
      {lines &&
        lines.length > 0 &&
        lines.map(line => {
          return (
            <li className="status-box-row pl-2" key={line}>
              {line}
            </li>
          );
        })}
    </ul>
  );
  return (
    <div className={`d-flex justify-content-between py-2 ${!last ? 'tv-table__row' : ''}`}>
      <div className="d-flex">
        <div className="d-flex flex-column justify-content-center">
          <CheckMark checked={ok} />
        </div>
        <div className="d-flex flex-column justify-content-center">
          <div className="mt-0 pt-1 pl-2">
            <b>{displayName}</b>
          </div>
          {formattedLines}
        </div>
      </div>
      <div className="d-flex flex-column justify-content-center mx-3 text-nowrap">
        {showAction && children ? children : ''}
      </div>
    </div>
  );
};

StatusBoxRow.defaultProps = {
  name: '',
  lines: [],
  showAction: false,
  ok: false,
  children: null,
  last: false,
};

StatusBoxRow.propTypes = {
  name: Types.oneOfType([Types.node, Types.string]),
  lines: Types.arrayOf(Types.oneOfType([Types.node, Types.string, Types.string])),
  showAction: Types.bool,
  ok: Types.bool,
  children: Types.node,
  last: Types.bool,
};

export default StatusBoxRow;
