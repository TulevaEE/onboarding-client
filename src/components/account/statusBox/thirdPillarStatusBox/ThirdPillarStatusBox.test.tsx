import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { ThirdPillarStatusBox } from './ThirdPillarStatusBox';
import StatusBoxRow from '../statusBoxRow';
import { activeThirdPillar, completeThirdPillarConversion, highFeeThirdPillar } from '../fixtures';

jest.mock('../../../common/apiHooks', () => ({
  useFundPensionStatus: () => ({ fundPensions: [] }),
}));

/**
 * @deprecated Use AccountPageView.test.tsx
 */
describe('ThirdPillarStatusBox', () => {
  let component: ShallowWrapper;
  const props = {
    conversion: completeThirdPillarConversion.thirdPillar,
    loading: false,
    thirdPillarFunds: [activeThirdPillar],
    thirdPillarActive: true,
  };

  beforeEach(() => {
    jest.useFakeTimers();
    jest.setSystemTime(new Date('2024-07-22T10:36:00Z'));
    component = shallow(<ThirdPillarStatusBox {...props} />);
  });

  it('renders the success flow', () => {
    expect(component).toMatchSnapshot();
  });

  it('does not show action when still loading', () => {
    component.setProps({ loading: true });
    expect(component.find(StatusBoxRow).prop('showAction')).toBeFalsy();
  });

  it('renders the "open third pillar" flow when user has no third pillar active', () => {
    component.setProps({ thirdPillarActive: false });
    expect(component).toMatchSnapshot();
  });

  it('renders the "pick tuleva" flow when user has some other fund manager', () => {
    component.setProps({
      conversion: {
        selectionPartial: false,
        selectionComplete: false,
        contribution: { yearToDate: 20, total: 20 },
        weightedAverageFee: 0.0049,
      },
    });
    expect(component).toMatchSnapshot();
  });

  it('renders the "pick tuleva" flow when partially converted with selection', () => {
    component.setProps({
      conversion: {
        selectionPartial: true,
        contribution: { yearToDate: 20, total: 20 },
        weightedAverageFee: 0.01,
      },
    });
    expect(component).toMatchSnapshot();
  });

  it('renders the "pick tuleva" flow when partially converted with transfer', () => {
    component.setProps({
      conversion: {
        transferPartial: true,
        contribution: { yearToDate: 20, total: 20 },
        weightedAverageFee: 0.0049,
      },
    });
    expect(component).toMatchSnapshot();
  });

  it('renders the "pick tuleva" flow when partially converted with transfer', () => {
    component.setProps({
      conversion: {
        transfersComplete: true,
        transfersPartial: true,
        selectionComplete: true,
        selectionPartial: true,
        contribution: { yearToDate: 20, total: 20 },
        weightedAverageFee: 0.01,
      },
    });
    expect(component).toMatchSnapshot();
  });

  it('renders the "transfer incomplete" flow when user has several funds', () => {
    component.setProps({
      conversion: {
        transfersComplete: false,
        transfersPartial: false,
        selectionComplete: true,
        selectionPartial: true,
        contribution: { yearToDate: 20, total: 20 },
        weightedAverageFee: 0.01,
      },
    });
    expect(component).toMatchSnapshot();
  });

  it('renders the "payment incomplete" flow when no funds have ever been transferred', () => {
    component.setProps({
      conversion: {
        transfersComplete: true,
        transfersPartial: true,
        selectionComplete: true,
        selectionPartial: true,
        contribution: { yearToDate: 0, total: 0 },
        weightedAverageFee: 0.01,
      },
    });
    expect(component).toMatchSnapshot();
  });

  it('renders the success flow when funds have not transferred yet but its only january', () => {
    jest.setSystemTime(new Date('2023-01-24T00:00:00'));
    component.setProps({
      conversion: {
        transfersComplete: true,
        transfersPartial: true,
        selectionComplete: true,
        selectionPartial: true,
        contribution: { yearToDate: 0, total: 20 },
        weightedAverageFee: 0.01,
      },
    });
    expect(component).toMatchSnapshot();
  });

  it('renders the "payment incomplete" flow when funds have not transferred yet and date between feb-nov', () => {
    jest.setSystemTime(new Date('2023-02-24T00:00:00'));
    component.setProps({
      conversion: {
        transfersComplete: true,
        transfersPartial: true,
        selectionComplete: true,
        selectionPartial: true,
        contribution: { yearToDate: 0, total: 20 },
        weightedAverageFee: 0.01,
      },
    });
    expect(component).toMatchSnapshot();
  });

  it('renders the december flow when date in december', () => {
    jest.setSystemTime(new Date('2023-12-24T00:00:00'));
    component.setProps({
      conversion: {
        transfersComplete: true,
        transfersPartial: true,
        selectionComplete: true,
        selectionPartial: true,
        contribution: { yearToDate: 0, total: 20 },
        weightedAverageFee: 0.01,
      },
    });
    expect(component).toMatchSnapshot();
  });

  it('renders the "pick tuleva" flow with error when in high cost fund', () => {
    component.setProps({
      conversion: {
        transferPartial: true,
        contribution: { yearToDate: 20, total: 20 },
        weightedAverageFee: 0.01,
      },
      thirdPillarFunds: [highFeeThirdPillar],
    });
    expect(component).toMatchSnapshot();
  });

  it('shows high fee notice when balance is zero but active fund has high fee', () => {
    component.setProps({
      conversion: {
        selectionPartial: false,
        selectionComplete: false,
        transfersPartial: false,
        transfersComplete: false,
        contribution: { yearToDate: 0, total: 0 },
        weightedAverageFee: 0,
      },
      thirdPillarFunds: [highFeeThirdPillar],
    });
    expect(component).toMatchSnapshot();
  });
});
