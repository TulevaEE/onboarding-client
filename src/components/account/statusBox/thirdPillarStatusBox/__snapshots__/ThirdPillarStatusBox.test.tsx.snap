// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThirdPillarStatusBox renders the "open third pillar" flow when user has no third pillar active 1`] = `
<StatusBoxRow
  error={true}
  lines={
    Array [
      <Memo(MemoizedFormattedMessage)
        id="account.status.choice.pillar.third.missing.label"
      />,
    ]
  }
  name={
    <Memo(MemoizedFormattedMessage)
      id="account.status.choice.pillar.third"
    />
  }
  showAction={true}
>
  <Link
    className="btn btn-primary"
    to="/3rd-pillar-flow"
  >
    <MemoizedFormattedMessage
      id="account.status.choice.pillar.third.missing.action"
    />
  </Link>
</StatusBoxRow>
`;

exports[`ThirdPillarStatusBox renders the "payment incomplete" flow when funds have not transferred yet and date between feb-nov 1`] = `
<StatusBoxRow
  lines={
    Array [
      <React.Fragment>
        <Memo(MemoizedFormattedMessage)
          id="account.status.choice.pillar.third.thisYearPaymentIncomplete.label"
        />
        <InfoTooltip
          name="third-pillar-tooltip"
        >
          <Memo(MemoizedFormattedMessage)
            id="account.status.choice.pillar.third.paymentInfo"
          />
        </InfoTooltip>
      </React.Fragment>,
      <Memo(Connect(ThirdPillarContributionAmount)) />,
    ]
  }
  name={
    <Memo(MemoizedFormattedMessage)
      id="account.status.choice.pillar.third"
    />
  }
  showAction={true}
  warning={true}
>
  <Link
    className="btn btn-primary"
    to="/3rd-pillar-payment"
  >
    <MemoizedFormattedMessage
      id="account.status.choice.pillar.third.success.action"
    />
  </Link>
</StatusBoxRow>
`;

exports[`ThirdPillarStatusBox renders the "payment incomplete" flow when no funds have ever been transferred 1`] = `
<StatusBoxRow
  error={true}
  lines={
    Array [
      <React.Fragment>
        <Memo(MemoizedFormattedMessage)
          id="account.status.choice.pillar.third.paymentIncomplete.label"
        />
        <InfoTooltip
          name="third-pillar-tooltip"
        >
          <Memo(MemoizedFormattedMessage)
            id="account.status.choice.pillar.third.paymentInfo"
          />
        </InfoTooltip>
      </React.Fragment>,
      <Memo(Connect(ThirdPillarContributionAmount)) />,
    ]
  }
  name={
    <Memo(MemoizedFormattedMessage)
      id="account.status.choice.pillar.third"
    />
  }
  showAction={true}
>
  <Link
    className="btn btn-primary"
    to="/3rd-pillar-payment"
  >
    <MemoizedFormattedMessage
      id="account.status.choice.pillar.third.success.action"
    />
  </Link>
</StatusBoxRow>
`;

exports[`ThirdPillarStatusBox renders the "pick tuleva" flow when partially converted with selection 1`] = `
<StatusBoxRow
  lines={
    Array [
      <Memo(MemoizedFormattedMessage)
        id="account.status.choice.pillar.third.transferIncomplete.label"
      />,
      <Memo(Connect(ThirdPillarContributionAmount)) />,
    ]
  }
  name={
    <Memo(MemoizedFormattedMessage)
      id="account.status.choice.pillar.third"
    />
  }
  showAction={true}
  warning={true}
>
  <Link
    className="btn btn-primary"
    to="/3rd-pillar-payment"
  >
    <MemoizedFormattedMessage
      id="account.status.choice.pillar.third.success.action"
    />
  </Link>
</StatusBoxRow>
`;

exports[`ThirdPillarStatusBox renders the "pick tuleva" flow when partially converted with transfer 1`] = `
<StatusBoxRow
  lines={
    Array [
      <React.Fragment>
        <Memo(MemoizedFormattedMessage)
          id="account.status.choice.lowFee.label"
        />
        <InfoTooltip
          name="third-pillar-tooltip"
        >
          <Memo(MemoizedFormattedMessage)
            id="account.status.choice.lowFee.description"
          />
        </InfoTooltip>
      </React.Fragment>,
      <Memo(Connect(ThirdPillarContributionAmount)) />,
    ]
  }
  name={
    <Memo(MemoizedFormattedMessage)
      id="account.status.choice.pillar.third"
    />
  }
  ok={true}
  showAction={true}
>
  <Link
    className="btn btn-light"
    to="/3rd-pillar-flow"
  >
    <MemoizedFormattedMessage
      id="account.status.choice.pillar.third.inactive.action"
    />
  </Link>
</StatusBoxRow>
`;

exports[`ThirdPillarStatusBox renders the "pick tuleva" flow when partially converted with transfer 2`] = `
<StatusBoxRow
  lines={
    Array [
      <Memo(MemoizedFormattedMessage)
        id="account.status.choice.lowFee.index.label"
      />,
      <Memo(Connect(ThirdPillarContributionAmount)) />,
    ]
  }
  name={
    <Memo(MemoizedFormattedMessage)
      id="account.status.choice.pillar.third"
    />
  }
  ok={true}
  showAction={true}
>
  <Link
    className="btn btn-primary"
    to="/3rd-pillar-payment"
  >
    <MemoizedFormattedMessage
      id="account.status.choice.pillar.third.success.action"
    />
  </Link>
</StatusBoxRow>
`;

exports[`ThirdPillarStatusBox renders the "pick tuleva" flow when user has some other fund manager 1`] = `
<StatusBoxRow
  lines={
    Array [
      <React.Fragment>
        <Memo(MemoizedFormattedMessage)
          id="account.status.choice.lowFee.label"
        />
        <InfoTooltip
          name="third-pillar-tooltip"
        >
          <Memo(MemoizedFormattedMessage)
            id="account.status.choice.lowFee.description"
          />
        </InfoTooltip>
      </React.Fragment>,
      <Memo(Connect(ThirdPillarContributionAmount)) />,
    ]
  }
  name={
    <Memo(MemoizedFormattedMessage)
      id="account.status.choice.pillar.third"
    />
  }
  ok={true}
  showAction={true}
>
  <Link
    className="btn btn-light"
    to="/3rd-pillar-flow"
  >
    <MemoizedFormattedMessage
      id="account.status.choice.pillar.third.inactive.action"
    />
  </Link>
</StatusBoxRow>
`;

exports[`ThirdPillarStatusBox renders the "pick tuleva" flow with error when in high cost fund 1`] = `
<StatusBoxRow
  error={true}
  lines={
    Array [
      <React.Fragment>
        <Memo(MemoizedFormattedMessage)
          id="account.status.choice.highFee.label"
        />
        <InfoTooltip
          name="third-pillar-tooltip"
        >
          <Memo(MemoizedFormattedMessage)
            id="account.status.choice.highFee.description"
          />
        </InfoTooltip>
      </React.Fragment>,
      <Memo(Connect(ThirdPillarContributionAmount)) />,
    ]
  }
  name={
    <Memo(MemoizedFormattedMessage)
      id="account.status.choice.pillar.third"
    />
  }
  showAction={true}
>
  <Link
    className="btn btn-primary"
    to="/3rd-pillar-flow"
  >
    <MemoizedFormattedMessage
      id="account.status.choice.choose.low.fees"
    />
  </Link>
</StatusBoxRow>
`;

exports[`ThirdPillarStatusBox renders the "transfer incomplete" flow when user has several funds 1`] = `
<StatusBoxRow
  lines={
    Array [
      <Memo(MemoizedFormattedMessage)
        id="account.status.choice.pillar.third.transferIncomplete.label"
      />,
      <Memo(Connect(ThirdPillarContributionAmount)) />,
    ]
  }
  name={
    <Memo(MemoizedFormattedMessage)
      id="account.status.choice.pillar.third"
    />
  }
  showAction={true}
  warning={true}
>
  <Link
    className="btn btn-primary"
    to="/3rd-pillar-payment"
  >
    <MemoizedFormattedMessage
      id="account.status.choice.pillar.third.success.action"
    />
  </Link>
</StatusBoxRow>
`;

exports[`ThirdPillarStatusBox renders the december flow when date in december 1`] = `
<StatusBoxRow
  error={true}
  lines={
    Array [
      <Memo(MemoizedFormattedMessage)
        id="account.status.choice.pillar.third.paymentDeadline"
        values={
          Object {
            "b": [Function],
          }
        }
      />,
      <Memo(Connect(ThirdPillarContributionAmount)) />,
    ]
  }
  name={
    <Memo(MemoizedFormattedMessage)
      id="account.status.choice.pillar.third"
    />
  }
  showAction={true}
>
  <Link
    className="btn btn-primary"
    to="/3rd-pillar-payment"
  >
    <MemoizedFormattedMessage
      id="account.status.choice.pillar.third.success.action"
    />
  </Link>
</StatusBoxRow>
`;

exports[`ThirdPillarStatusBox renders the success flow 1`] = `
<StatusBoxRow
  lines={
    Array [
      <Memo(MemoizedFormattedMessage)
        id="account.status.choice.lowFee.index.label"
      />,
      <Memo(Connect(ThirdPillarContributionAmount)) />,
    ]
  }
  name={
    <Memo(MemoizedFormattedMessage)
      id="account.status.choice.pillar.third"
    />
  }
  ok={true}
  showAction={true}
>
  <Link
    className="btn btn-primary"
    to="/3rd-pillar-payment"
  >
    <MemoizedFormattedMessage
      id="account.status.choice.pillar.third.success.action"
    />
  </Link>
</StatusBoxRow>
`;

exports[`ThirdPillarStatusBox renders the success flow when funds have not transferred yet but its only january 1`] = `
<StatusBoxRow
  lines={
    Array [
      <Memo(MemoizedFormattedMessage)
        id="account.status.choice.lowFee.index.label"
      />,
      <Memo(Connect(ThirdPillarContributionAmount)) />,
    ]
  }
  name={
    <Memo(MemoizedFormattedMessage)
      id="account.status.choice.pillar.third"
    />
  }
  ok={true}
  showAction={true}
>
  <Link
    className="btn btn-outline-primary"
    to="/3rd-pillar-payment"
  >
    <MemoizedFormattedMessage
      id="account.status.choice.pillar.third.success.action"
    />
  </Link>
</StatusBoxRow>
`;
